leafmachine-demo/main.py

```diff
@@ -0,0 +1,172 @@
+import json
+import logging
+import os
+import uuid
+import requests
+from typing import Tuple, Any, Dict, List
+from kafka import KafkaConsumer, KafkaProducer
+import shared
+
+logging.basicConfig(format="%(asctime)s - %(message)s", level=logging.INFO)
+
+
+def start_kafka() -> None:
+    """
+    Start a kafka listener and process the messages by unpacking the image.
+    When done it will republish the object, so it can be validated and stored by the processing service.
+    :param predictor: The predictor which will be used to run the plant organ segmentation
+
+    """
+    consumer = KafkaConsumer(
+        os.environ.get("KAFKA_CONSUMER_TOPIC"),
+        group_id=os.environ.get("KAFKA_CONSUMER_GROUP"),
+        bootstrap_servers=[os.environ.get("KAFKA_CONSUMER_HOST")],
+        value_deserializer=lambda m: json.loads(m.decode("utf-8")),
+        enable_auto_commit=True,
+    )
+
+    producer = KafkaProducer(
+        bootstrap_servers=[os.environ.get("KAFKA_PRODUCER_HOST")],
+        value_serializer=lambda m: json.dumps(m).encode("utf-8"),
+    )
+
+    for msg in consumer:
+        logging.info(f"Received message: {str(msg.value)}")
+        json_value = msg.value
+        try:
+            shared.mark_job_as_running(job_id=json_value.get("jobId"))
+            digital_object = json_value.get("object")
+            additional_info_annotations, image_height, image_width = run_leafmachine(digital_object.get("ac:accessURI"))
```

Comment:
If no results are found, you should still send an annotation with a comment to inform the user what has happened. See `shared.map_to_empty_annotation`.